integration/test/Testlib/App.hs

```diff
@@ -0,0 +1,935 @@
+-------------------------------------------------------------------------------
+
+yellow :: String
+yellow = "\x1b[38;5;11m"
+
+blue :: String
+blue = "\x1b[38;5;6m"
+
+purpleish :: String
+purpleish = "\x1b[38;5;13m"
+
+orange :: String
+orange = "\x1b[38;5;3m"
+
+red :: String
+red = "\x1b[38;5;1m"
+
+resetColor :: String
+resetColor = "\x1b[0m"
+
+colored :: String -> String -> String
+colored color s = color <> s <> resetColor
+
+indent :: Int -> String -> String
+indent n s =
+  unlines (map (pad <>) (lines s))
+  where
+    pad = replicate n ' '
+
+hline :: String
+hline = replicate 40 '-'
+
+-------------------------------------------------------------------------------
+-- - SECTION_MODSERVICE : Start services with modified configuration
+-------------------------------------------------------------------------------
+
+withModifiedService ::
+  Service ->
+  -- | function that edits the config
+  (Value -> App Value) ->
+  -- | This action wil access the modified spawned service
+  App a ->
+  App a
+withModifiedService srv modConfig k = do
+  withModifiedServices (Map.singleton srv modConfig) k
+
+withModifiedServices :: Map Service (Value -> App Value) -> App a -> App a
+withModifiedServices services k = do
+  ports <- Map.traverseWithKey (\_ _ -> liftIO openFreePort) services
+
+  let updateServiceMapInConfig :: Value -> App Value
+      updateServiceMapInConfig config =
+        foldlM
+          ( \c (srv, (port, _)) ->
+              c
+                & setField
+                  (serviceName srv)
+                  ( object
+                      [ "host" .= ("127.0.0.1" :: String),
+                        "port" .= port
+                      ]
+                  )
+          )
+          config
+          (Map.assocs ports)
+
+  instances <- for (Map.assocs services) $ \(srv, modifyConfig) -> do
+    basedir <- asks (.serviceConfigsDir)
+    let srvName = serviceName srv
+        cfgFile = basedir </> srvName </> "conf" </> (srvName <> ".yaml")
+    config <- do
+      eith <- liftIO (Yaml.decodeFileEither cfgFile)
+      case eith of
+        Left err -> failApp ("Error while parsing " <> cfgFile <> ": " <> Yaml.prettyPrintParseException err)
+        Right value -> pure value
+    config' <- updateServiceMapInConfig config >>= modifyConfig
+    (tempFile, fh) <- liftIO $ openBinaryTempFile "/tmp" (srvName <> ".yaml")
+    liftIO $ BS.hPut fh (Yaml.encode config')
+    hClose fh
+
+    (cwd, exe) <-
+      asks (.servicesCwdBase) <&> \case
+        Nothing -> (Nothing, srvName)
+        Just dir ->
+          (Just (dir </> srvName), "./dist" </> srvName)
+
+    (port, socket) <- maybe (failApp "the impossible in withServices happened") pure (Map.lookup srv ports)
+    liftIO $ N.close socket
+    (_, _, _, ph) <- liftIO $ createProcess (proc exe ["-c", tempFile]) {cwd = cwd}
+    pure ph
+
+  let stopInstances = liftIO $ do
+        -- when running from repl this hangs for 30 seconds
```

Comment:
is this still true?  (the before of https://github.com/wireapp/wire-server/pull/3171/commits/022a4439f5bf63a4454b3579365862aa10d4e90e looks weird...)

Reply:
No, fixed the comment in 60451d9